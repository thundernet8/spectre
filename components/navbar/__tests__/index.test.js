import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import NavBar from "..";

const NavSection = NavBar.Section;

describe("NavBar", () => {
    it("should render NavBar and NavSection", () => {
        const navbar = mount(
            <NavBar>
                <NavSection>Left section</NavSection>
                <NavSection center>Center section</NavSection>
                <NavSection>Right section</NavSection>
            </NavBar>
        );

        // expect(navbar.find("section").get(0).tagName).toEqual("header");

        expect(navbar.hasClass("spt-navbar")).toBe(true);

        expect(navbar.find("section").get(0).tagName).toEqual("SECTION");

        expect(navbar.childAt(0).hasClass("spt-navbar-section")).toBe(true);

        expect(navbar.childAt(1).hasClass("spt-navbar-center")).toBe(true);
    });

    it("renders correctly", () => {
        const tree = renderer
            .create(
                <NavBar>
                    <NavSection>Left section</NavSection>
                    <NavSection center>Center section</NavSection>
                    <NavSection>Right section</NavSection>
                </NavBar>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
