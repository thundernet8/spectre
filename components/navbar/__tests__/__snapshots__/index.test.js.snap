// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<header
  className="spt-navbar"
  style={undefined}
>
  <section
    className="spt-navbar-section"
    style={undefined}
  >
    Left section
  </section>
  <section
    className="spt-navbar-center"
    style={undefined}
  >
    Center section
  </section>
  <section
    className="spt-navbar-section"
    style={undefined}
  >
    Right section
  </section>
</header>
`;
